src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -156,11 +156,18 @@ public CheckResult SetBindingInfo(RecordType parameterType)
         }
 
         private FormulaType _expectedReturnType;

```

Comment:
>_expectedReturnType [](http://example.com/codeflow?start=28&length=19)

can we remove this field, since the new Func<> covers it now. 

And we could rename:
_expectedCoerceToType

to something more general - the function can do any check, not just recursion.  For example, it may check that it's a record with a certain required field. 

So maybe just: 
_expectedReturnTypeValidator

Reply:
update this.